subsys/mpsl/fem/nrf21540_gpio/mpsl_fem_nrf21540_gpio.c

```diff
@@ -298,116 +308,131 @@ static int fem_nrf21540_gpio_configure(void)
 }
 #endif
 
-#if IS_ENABLED(CONFIG_MPSL_FEM_SIMPLE_GPIO)
-static int fem_simple_gpio_configure(void)
+static int mpsl_fem_init(const struct device *dev)
 {
-#if !DT_NODE_EXISTS(DT_NODELABEL(nrf_radio_fem))
-#error Node with label 'nrf_radio_fem' not found in the devicetree.
-#endif
-	int err;
+	ARG_UNUSED(dev);
 
-#if DT_NODE_HAS_PROP(DT_NODELABEL(nrf_radio_fem), ctx_gpios)
-	uint8_t ctx_gpiote_channel;
+	mpsl_fem_device_config_254_apply_set(IS_ENABLED(CONFIG_MPSL_FEM_DEVICE_CONFIG_254));
 
-	if (nrfx_gpiote_channel_alloc(&ctx_gpiote_channel) != NRFX_SUCCESS) {
-		return -ENOMEM;
+	return fem_nrf21540_gpio_configure();
+}
+
+SYS_INIT(mpsl_fem_init, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
+
+#else /* !defined(CONFIG_MPSL_FEM_PIN_FORWARDER) */
+
+static void fem_pin_num_correction(uint8_t *p_gpio_pin, const char *gpio_lbl)
```

Comment:
This function will be exactly the same for every FEM type. Maybe it should be extracted to a separate "utilities" file?

Reply:
In general some functions are duplicated and they could be made common. I left them duplicated on purpose - I wanted the implementations to be completely independent. Every implementation must provide everything _it_ needs - nothing more, nothing less. We could decrease number of repetitions at the cost of slightly compromising the goal of fully independent FEM implementations that can be added without any modifications to the common code. What are your opinions on this @ahasztag @ankuns?

Reply:
In case of `fem_pin_num_correction` the FEMs still remain fully independent - this is simply a calculator which simply converts a pin + label to a single number. And if new hardware platforms which would have more gpio ports are added, keeping it in seperate files would cause the need to introduce the change in all possible FEM implementations

Reply:
Yeah, that makes sense.